### PR TITLE
[DIR-1608] Display error message on errors when fetching logs

### PR DESCRIPTION
**Pull request by sebxian**
## Description

Previously, the ui just didn't render any logs when fetching logs failed, e.g. with a 500 server error. With this fix, an error message is rendered when a request returns an error.

### Notes

Implementation:
* This will display error messages when an error response is received during the non-streaming request. I don't know if an error via streaming would end up being returned by useLogs() in the same way and according to Alexander, the streaming API currently will not return errors (instead, it will just terminate the connection when something fails). I have created DIR-1615 to look at this in more detail.
* I created a component ApiError that should be reusable on every page where we want to pull the `error` prop from a useQuery hook and display an error, so we don't need to implement typesafe parsing of the error etc. on every page.
* It is possible to define a custom label for the error message on every page to make it context dependent, but there is also a fallback default label if no label is provided. Additionally, the alert will render human readable error message from the API response (if present).
* I updated the ApiErrorSchema. It will now include the http status response code, and if present, the response body under "body" (previously "json"). The original response will not be included (we never used it and it would be complicated to do so). I also updated our documentation/comments in errorHandling.ts.

Test notes:
* I added a test that mocks an error response for the logs on monitoring.
* I added a test to mock an error response for instance logs as well, and added a note in DIR-1595 to integrate it with the currently broken tests when they are fixed.

Review notes:
To fake an error outside of the tests, you could hard code this in `src/pages/namespace/Monitoring/Logs/index.tsx` instead of the useQuery error:

```
  const error: QueryErrorType = {
    status: 422,
    body: { code: 422, message: "oh no!" },
  };
```

Layout notes:
In theory, it is possible to have logs as well as errors (e.g., when only one of multiple requests fails). So I covered both cases in development (but only with fake/mock errors).

Screenshot with full and empty logs below. At first glance there do not seem to be problems with the height of the virtualizer container, but I haven't tested this extensively. I see a risk that there will be problems when an error appears after the list has already loaded (but that seems like an unlikely edge case).

image

image

## Checklist

- [x] Documentation updated if required
- [x] Test coverage is appropriate
      
## Checklist Internal

- [x] Linear issue linked (e.g. [DIR-XXXX] pull request title)
- [x] Has the PR been labeled

### CODE REVIEW

**Review by stefan-kracht — CHANGES_REQUESTED** [review]
Very nice. The E2E tests helped me a lot with doing the review!

File: ui/src/pages/namespace/Instances/Detail/Main/Logs/index.tsx [hunk 1]
```diff
@@ -117,6 +118,11 @@ const LogsPanel = () => {
           </TooltipProvider>
         </ButtonBar>
       </div>
+      <div className="mb-3">
```
[the remark this answers is not in the document]
Comment (sebxian) [reply ui/src/pages/namespace/Instances/Detail/Main/Logs/index.tsx]: This was tricky to fix due to the grid layout and the requirements of the virtual scrolling container, but as discussed in person, it could be [solved](a87b50e49) by moving the alert component into the header of the grid. Thanks for the support!

Screenshots:
image
image
image

[the remark this answers is not in the document]
Comment (sebxian) [reply ui/src/pages/namespace/Instances/Detail/Main/Logs/index.tsx]: As discussed in person, this could be solved (a87b50e49) by moving the ApiError component into the header of the grid layout.
image
image
image

File: ui/src/components/ApiError/index.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+const ApiError = ({
+  error,
+  label,
+}: {
+  error: QueryErrorType;
+  label?: string;
+}) => {
+  const { t } = useTranslation();
+  const errorMessage = getMessageFromApiError(error);
```
[the remark this answers is not in the document]
Comment (sebxian) [reply ui/src/components/ApiError/index.tsx:14]: I added different messages for the different use cases with and without an error message in 5ebdc18c7. To avoid an explosion in complexity, I decided to remove the ability to customize the error message. Now the component is a bit less customizable, but easy to use (otherwise we would need to consider messages for the different use cases in every parent component).

**Review by stefan-kracht — APPROVED** [review]
